### PR TITLE
#1756 generate 3-letter airport codes

### PR DESCRIPTION
**Pull request by asolntsev**
A.k.a. IATA code (passenger-friendly 3-letter code).

This commit adds
1. enum `AirportCodeType (IATA | ICAO)`
2. method `aviation.airport(IATA)` -> 3-letter code
3. method `aviation.airport(ICAO)` -> 4-letter code

### CODE REVIEW

File: src/main/java/net/datafaker/providers/base/Aviation.java [hunk 1]
```diff
@@ -104,17 +121,27 @@ public String engineType() {
...
-     *
-     * @return A random flight number with IATA or ICAO format in a String.
+     * @deprecated use {@link #flight(AirportCodeType)} instead
      */
+    @Deprecated
     public String flight(String type) {
-        String airline;
         if ("ICAO".equalsIgnoreCase(type)) {
-            airline = resolve("aviation.ICAO_airline");
+            return flight(AirportCodeType.ICAO);
```
Comment (kingthorin) [inline src/main/java/net/datafaker/providers/base/Aviation.java]: Why qualified here but not at 131?

File: src/main/java/net/datafaker/providers/base/Aviation.java [hunk 2]
```diff
@@ -57,12 +59,27 @@ public String civilHelicopter() {
 
     /**
      * Returns an airport ICAO code.
-     * See also: <a href="https://en.wikipedia.org/wiki/List_of_airports_by_ICAO_code:_A">https://en.wikipedia.org/wiki/List_of_airports_by_ICAO_code:_A</a>
+     * <p>
+     *  Seems that this method can return some local airport codes in addition to official ICAO code.
+     *  For example, it returns "CNT3" code for Ogoki Post Airport, but its official ICAO code is "CYKP".
+     * </p>
+     * @see <a href="https://en.wikipedia.org/wiki/List_of_airports_by_ICAO_code:_A">https://en.wikipedia.org/wiki/List_of_airports_by_ICAO_code:_A</a>
```
[the remark this answers is not in the document]
Comment (asolntsev) [reply src/main/java/net/datafaker/providers/base/Aviation.java:67]: Yes, I left the initial description and added my observation.

File: src/main/java/net/datafaker/providers/base/Aviation.java [hunk 3]
```diff
@@ -104,17 +121,27 @@ public String engineType() {
...
+     *
+     * @return A random flight number with IATA or ICAO format in a String.
+     */
+    public String flight(AirportCodeType type) {
+        String airline = switch (type) {
+            case ICAO -> resolve("aviation.ICAO_airline");
+            case IATA -> resolve("aviation.IATA_airline");
+        };
         int number = faker.number().numberBetween(0, 9999);
         return airline + number;
```
[the remark this answers is not in the document]
Comment (asolntsev) [reply src/main/java/net/datafaker/providers/base/Aviation.java]: Null is not allowed. 
If user passes null, it will throw a readable exception:
> java.lang.NullPointerException: Cannot invoke "net.datafaker.providers.base.AirportCodeType.ordinal()" because "type" is null

No need to fix anything.

File: src/main/java/net/datafaker/providers/base/Aviation.java (hunk 1 above)
Comment (asolntsev) [reply src/main/java/net/datafaker/providers/base/Aviation.java]: Ooops, fixed.

File: src/main/java/net/datafaker/providers/base/Aviation.java [hunk 4]
```diff
@@ -57,12 +59,27 @@ public String civilHelicopter() {
...
 
+    /**
+     * Returns either 3-letter IATA code or 4-letter ICAO code of a random airport
+     * @see AirportCodeType
+     */
+    public String airport(AirportCodeType codeType) {
+        return switch (codeType) {
+            case IATA -> resolve("aviation.airport_iata");
+            case ICAO -> resolve("aviation.airport_icao");
+        };
```
[the remark this answers is not in the document]
Comment (asolntsev) [reply src/main/java/net/datafaker/providers/base/Aviation.java:81]: Null is not allowed. 
If user passes null, it will throw a readable exception:
> java.lang.NullPointerException: Cannot invoke "net.datafaker.providers.base.AirportCodeType.ordinal()" because "type" is null

No need to fix anything.

Also, I added a test for `airport((AviationCodeType) null)`.

File: src/test/java/net/datafaker/providers/base/AviationTest.java [hunk 5]
```diff
@@ -45,9 +60,32 @@ void gate() {
...
+        assertThat(aviation.airport(IATA))
+            .as("IATA is 3-letter code consisting of Latin letters")
+            .hasSize(3).matches("[A-Z]{3}");
+    }
+
     @Override
     protected Collection<TestSpec> providerListTest() {
         return List.of(TestSpec.of(aviation::airport, "aviation.airport", "\\w{4}"),
+            TestSpec.of(() -> aviation.airport(IATA), "aviation.airport_iata", "[A-Z]{3}"),
+            TestSpec.of(() -> aviation.airport(ICAO), "aviation.airport_icao", "[A-Z]{4}"),
```
Comment (kingthorin) [inline src/test/java/net/datafaker/providers/base/AviationTest.java]: Match or re-use the constants from earlier?

Comment (kingthorin) [reply src/test/java/net/datafaker/providers/base/AviationTest.java]: If these are correct, simplify the earlier ones?

Comment (asolntsev) [reply src/test/java/net/datafaker/providers/base/AviationTest.java]: Good point. I've extracted these regexes to constants.
